torchtitan/protocols/hf_adapter.py

```diff
@@ -0,0 +1,42 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+from abc import ABC, abstractmethod
+from typing import Any
+
+import torch.nn as nn
+
+
+class HFAdapter(ABC):
```

Comment:
Is there a reason why we don't define a more general model adapter? Like if users would like to load from the Meta llama 4 definition?